langchain/document_loaders/directory.py

```diff
@@ -48,11 +50,29 @@ def __init__(
         self.silent_errors = silent_errors
         self.recursive = recursive
         self.show_progress = show_progress
+        self.use_multithreading = use_multithreading
+
+    def load_file(
+        self, item: Path, path: Path, docs: List[Document], pbar: Optional[Any]
+    ) -> None:
+        if item.is_file():
+            if _is_visible(item.relative_to(path)) or self.load_hidden:
```

Comment:
nit, maybe use early-return?
```python3
if not item.is_file():
  return
if not …:
  return 
try:
  …
...
```

Reply:
I didn't change this code, just wrapped it into a function. I only really added this condition:
```
        if self.use_multithreading:
            with concurrent.futures.ThreadPoolExecutor(max_workers=4) as executor:
                executor.map(lambda i: self.load_file(i, p, docs, pbar), items)
        else:
            for i in items:
                self.load_file(i, p, docs, pbar)
```